examples/howto/Model_Builder.myst.md

```diff
@@ -0,0 +1,255 @@
+    version = "0.1"
+
+    def build_model(self, model_config, data=None):
+        """
+        build_model creates the model and loads it to memory stack.
+
+        Parameters:
+        model_config: dictionary
+            it is a dictionary with all the parameters of pm.model that we need in our model example:  a_loc, a_scale, b_loc
+        data: Dict[str, Union[np.ndarray, pd.DataFrame, pd.Series]]
+            Default value is None. It is the data we need in our model to fit and other purposes, data can be passed in fit as well.
+        """
+
+        if data is not None:
+            x = pm.MutableData("x", data["input"].values)
+            y_data = pm.MutableData("y_data", data["output"].values)
+
+        # prior parameters
+        a_loc = model_config["a_loc"]
+        a_scale = model_config["a_scale"]
+        b_loc = model_config["b_loc"]
+        b_scale = model_config["b_scale"]
+        obs_error = model_config["obs_error"]
+
+        # priors
+        a = pm.Normal("a", a_loc, sigma=a_scale)
+        b = pm.Normal("b", b_loc, sigma=b_scale)
+        obs_error = pm.HalfNormal("σ_model_fmc", obs_error)
+
+        # observed data
+        if data is not None:
+            y_model = pm.Normal("y_model", a + b * x, obs_error, shape=x.shape, observed=y_data)
+
+    def _data_setter(self, data: pd.DataFrame):
+        """
+        _data_setter works as a set_data for the model and updates the data whenever we need to.
+        Parameters:
+        data: Dict[str, Union[np.ndarray, pd.DataFrame, pd.Series]]
+            It is the data we need to update for the model.
+        """
+
+        with self.model:
+            pm.set_data({"x": data["input"].values})
+            if "output" in data.columns:
+                pm.set_data({"y_data": data["output"].values})
+
+    @classmethod
+    def create_sample_input(cls):
+        """
+        create_sample_input creates the sample input we need for our model, user can either use this or initialise their own parameters accordingly.
+        """
+
+        x = np.linspace(start=0, stop=1, num=100)
+        y = 5 * x + 3
+        y = y + np.random.normal(0, 1, len(x))
+        data = pd.DataFrame({"input": x, "output": y})
+
+        model_config = {
+            "a_loc": 0,
+            "a_scale": 10,
+            "b_loc": 0,
+            "b_scale": 10,
+            "obs_error": 2,
+        }
+
+        sampler_config = {
+            "draws": 1_000,
+            "tune": 1_000,
+            "chains": 3,
+            "target_accept": 0.95,
+        }
+
+        return data, model_config, sampler_config
+```
+
+Now we can create an object of LinearModel type, which we can edit according to our use or use the default model as defined by the user.
+Most importantly, if you make _ a really cool model_ and want to deploy the same, it will be easier for you to make a class and share the model so people can use it via the object instead of redefining the model every time they need it. <br>
```

Comment:
```suggestion
Most importantly, if you make _a really cool model_ and want to deploy the same, it will be easier for you to make a class and share the model so people can use it via the object instead of redefining the model every time they need it. <br>
```

There is an extra space which makes it not render properly. Current look:

![imatge](https://user-images.githubusercontent.com/23738400/213527643-afb49eb0-66c1-4805-9f33-8024853c8136.png)